astronomy/standard_product_3.hpp

```diff
@@ -0,0 +1,94 @@
+﻿#pragma once
+
+#include <filesystem>
+#include <map>
+#include <optional>
+#include <string>
+
+#include "astronomy/frames.hpp"
+#include "geometry/named_quantities.hpp"
+
+namespace principia {
+namespace astronomy {
+namespace internal_standard_product_3 {
+
+using geometry::Instant;
+using geometry::Position;
+using geometry::Velocity;
+
+// A representation of data in the extended standard product 3 orbit format.
+// Specification:
+// - version a: ftp://igs.org/pub/data/format/sp3_docu.txt;
+// - version b: ftp://igs.org/pub/data/format/sp3_glon.txt.
+// - version c: ftp://igs.org/pub/data/format/sp3c.txt.
+// - version d: ftp://igs.org/pub/data/format/sp3d.pdf.
+class StandardProduct3 {
+ public:
+  enum class Dialect {
+    // SP3 file conformant with the specification.
+    Standard,
+    // International Laser Ranging Service official combined product ILRSA.
+    // Divergences from the specification:
+    // - comments start with %/* instead of /*;
+    // - missing last line EOF.
+    ILRSA,
+    // International Laser Ranging Service official backup combined product
+    // ILRSB.
+    // Divergences from the specification:
+    // - the number of epochs given on the first line is off by one;
+    // - comments start with %/* instead of /*;
+    // - the fields of the epoch record are shifted one column to the left;
+    // - the minute field of the epoch record takes the value 60 at the end of
+    //   the hour.
+    ILRSB,
+  };
+
+  enum SatelliteGroup : char {
+    General = 'L',
+    GPS = 'G',
+    ГЛОНАСС = 'R',
+    Galileo = 'E',
+    北斗 = 'C',
+    準天頂衛星 = 'J',
+    IRNSS = 'I',
+  };
+
+  struct SatelliteIdentifier {
+    SatelliteGroup group;
+    int index;
+
+    friend bool operator<(SatelliteIdentifier const& left,
+                          SatelliteIdentifier const& right) {
+      return left.group < right.group ||
+             (left.group == right.group && left.index < right.index);
+    }
+  };
+
+  struct OrbitPoint {
```

Comment:
`OrbitDegreesOfFreedom`?  The word "point" has a specific technical meaning in our code base.

Reply:
Making it a DiscreteTrajectory.